metadata-ingestion/src/datahub/api/entities/corpgroup/corpgroup.py

```diff
@@ -39,46 +54,183 @@ class CorpGroup:
+                    email=self.email,
+                    description=self.description,
+                    slack=self.slack,
+                )
+            )
+            # picture link is only available in the editable aspect, so we have to use it if it is provided
+            if self._needs_editable_aspect():
+                aspects.append(
+                    CorpGroupEditableInfoClass(
+                        description=self.description,
+                        pictureLink=self.picture_link,
+                        slack=self.slack,
+                        email=self.email,
+                    )
+                )
+
+        for aspect in aspects:
+            yield MetadataChangeProposalWrapper(entityUrn=self.urn, aspect=aspect)
+
+        # Unfortunately, admins and members fields in CorpGroupInfo has been deprecated
+        # So we need to emit Ownership and GroupMembership oriented to the individual users
+        # TODO: Move this to PATCH MCP-s once these aspects are supported via patch.
+        if generation_config.datahub_graph is not None:
+            datahub_graph = generation_config.datahub_graph
+            for urn in admin_urns:
+                ownership = datahub_graph.get_aspect(
+                    urn, OwnershipClass
+                ) or OwnershipClass(owners=[])
+                if self.urn not in [owner.owner for owner in ownership.owners]:
+                    ownership.owners = ownership.owners + [
+                        OwnerClass(owner=urn, type=OwnershipTypeClass.TECHNICAL_OWNER)
+                    ]
+                    yield MetadataChangeProposalWrapper(
+                        entityUrn=self.urn, aspect=ownership
+                    )
+
+            for urn in member_urns:
+                group_membership = datahub_graph.get_aspect(
+                    urn, GroupMembershipClass
+                ) or GroupMembershipClass(groups=[])
+                if self.urn not in group_membership.groups:
+                    group_membership.groups = sorted(
+                        set(group_membership.groups + [self.urn])
+                    )
+                    yield MetadataChangeProposalWrapper(
+                        entityUrn=urn, aspect=group_membership
+                    )
+        else:
+            if admin_urns or member_urns:
+                logger.error(
+                    "Unable to emit group ownership because a DataHubGraph instance was not provided. Continuing to preserve backwards compatibility with existing functionality."
+                )
+
+        # emit status aspects for all user urns referenced (to ensure they get created)
+        for urn in set(admin_urns).union(set(member_urns)):
+            yield MetadataChangeProposalWrapper(
+                entityUrn=urn, aspect=StatusClass(removed=False)
+            )
+
+    @lru_cache(maxsize=32)
+    def _datahub_graph_from_datahub_rest_emitter(
+        self, rest_emitter: DatahubRestEmitter
+    ) -> DataHubGraph:
```

Comment:
I feel like this should be a method of the DataHubGraph class

Reply:
If this was a long term function, then yes. Since this is a specific workaround for the case where corpgroup is being used in the current scenario with a rest emitter, and until we get patch support, I feel like this function should be scoped here.